model/filterDecorator.php

```diff
@@ -0,0 +1,8 @@
+<?php 
+require_once 'filter.php';
+abstract class FilterDecorator extends Filter
+{
+        private $filter;
+    abstract public function generate();

```

Comment:
The abstract function was already declared in `Filter`, there's no need to declare it again here.